packages/data/src/data-manager.ts

```diff
@@ -367,9 +369,8 @@ export class DataManager implements DataManagerInterface {
         // If there are some matched audiences
         if (
           !visitorProperties ||
-          matchedAudiences.length ||
-          matchedSegmentations.length ||
-          !audiences.length // Empty audiences list means there's no restriction for the audience
+          (audiencesToCheck.length && matchedAudiences.length) || // Empty audiences list means there's no restriction for the audience
```

Comment:
you are right @abbaseya 

probably should be like 
`!visitorProperties ||
  !audiences.length 
  (audiencesToCheck.length && matchedAudiences.length) ||
  (segments.length && matchedSegments.length)`

also, two notes:

1. I think it would be clearer if the condition here would just say: 
`audiencesMatched && segmentsMatched` and making sure those two flags are correctly set above
2. I see we will consider locations matched if `!locationProperties` , same for audiences/segmetns if 
`!visitorProperties` . Any reason for that? It does not make sense to me to default to matching in case data to match against is missing :)